draft-ietf-ppm-dap.md

```diff
@@ -814,29 +945,56 @@ structured as follows:
 
 ~~~
 struct {
-  Nonce nonce;
-  Extension extensions<0..2^16-1>;
+  ReportInfo info;
   HpkeCiphertext encrypted_input_share;
 } ReportShare;
 
 struct {
   TaskID task_id;
   AggregationJobID job_id;
   opaque agg_param<0..2^16-1>;
+  select (query_type) { // determined by task configuration
+    time-interval: Empty;
+    fixed-size: BatchId batch_id;
+  }
   ReportShare report_shares<1..2^32-1>;
 } AggregateInitializeReq;
 ~~~
 
-[[OPEN ISSUE: consider sending report shares separately (in parallel) to the aggregate instructions. RIght now, aggregation parameters and the corresponding report shares are sent at the same time, but this may not be strictly necessary. ]]
+[[OPEN ISSUE: Consider sending report shares separately (in parallel) to the
+aggregate instructions. Right now, aggregation parameters and the corresponding
+report shares are sent at the same time, but this may not be strictly
+necessary.]]
+
+This message consists of:
 
-The `nonce` and `extensions` fields of each ReportShare match that in the Report
-uploaded by the client. The `encrypted_input_share` field is the `HpkeCiphertext`
-whose index in `Report.encrypted_input_shares` is equal to the index of the aggregator
-in the task's `aggregator_endpoints` to which the AggregateInitializeReq is being sent.
-The `agg_param` field is an opaque, VDAF-specific aggregation parameter provided during a collection flow.
-The `job_id` parameter contains the leader's chosen AggregationJobID.
+* The ID of the task for which the reports will be aggregated.
 
-[[OPEN ISSUE: Check that this handling of `agg_param` is appropriate when the definition of Poplar is done.]]
+* The aggregation job ID.
+
+* The opaque, VDAF-specific aggregation parameter provided during the
+  collection flow (`agg_param`),
+
+  [[OPEN ISSUE: Check that this handling of `agg_param` is appropriate when the
+  definition of Poplar is done.]]
+
+* Information used by the Aggregators to determine how to aggregate each report:
+
+    * For fixed-size tasks, the Leader specifies a "batch ID" that determines
```

Comment:
Is it worth mentioning that the binding of batch_id and agg_job_ids is permanent? so collect query can be idempotent: query of the same batch_id should return the same result, if max_batch_lifetime > 0.

Reply:
Maybe, but not here. This would go in the collect sub-protocol I'd think.